From 93c5e68ab2eb5915cf75551e8f110de63b4ded9c Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:19:37 -0400
Subject: [iommu] vt-d: Downgrade the warning if enabling irq remapping fails

Message-id: <20140810181937.24055.84612.stgit@gir.stowe>
Patchwork-id: 87572
O-Subject: [RHEL7.1 PATCH 01/19] iommu/vt-d: Downgrade the warning if enabling irq remapping fails
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: d01140df6a32554728ab055f2d82f8028a73d2f5

commit d01140df6a32554728ab055f2d82f8028a73d2f5
Author: Andy Lutomirski <luto@amacapital.net>
Date:   Mon May 13 15:22:42 2013 -0700

    iommu/vt-d: Downgrade the warning if enabling irq remapping fails

    This triggers on a MacBook Pro.
    See https://bugzilla.redhat.com/show_bug.cgi?id=948262 for
    the problem report.

    Signed-off-by: Andy Lutomirski <luto@amacapital.net>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel_irq_remapping.c b/drivers/iommu/intel_irq_remapping.c
index 45011f6..b97d70b 100644
--- a/drivers/iommu/intel_irq_remapping.c
+++ b/drivers/iommu/intel_irq_remapping.c
@@ -665,8 +665,7 @@ error:
   */
 
  if (x2apic_present)
-  WARN(1, KERN_WARNING
-   "Failed to enable irq remapping.  You are vulnerable to irq-injection attacks.\n");
+  pr_warn("Failed to enable irq remapping.  You are vulnerable to irq-injection attacks.\n");
 
  return -1;
 }
-- 
1.7.1